media-libs/virglrenderer: wire up venus,vaapi,drm-renderers

mm1ke opened the pull request:
![20250630_20h33m47s_grim](https://github.com/user-attachments/assets/b4e8e137-fa46-43c6-bc4d-e4f45302c909)

Honestly, that's pretty wild. I didn't expect this would work basically out of the box.
Playing a game inside a virtual machine without any gpu passthrough with basically host performance :)


This PR updates `media-libs/virglrenderer` to enable `venus`, `vaapi` and `drm-renderers`.

So far i only tested the `venus` and ` vaapi` part. `venus` works amazingly well (as seen in the screenshot).
`vaapi` does _something_. `vainfo` for example shows following:
```
Trying display: wayland
libva info: VA-API version 1.22.0
libva info: Trying to open /usr/lib64/va/drivers/virtio_gpu_drv_video.so
libva info: Found init function __vaDriverInit_1_22
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.22 (libva 2.22.0)
vainfo: Driver version: Mesa Gallium driver 25.1.4 for virgl (AMD Radeon RX 7900 XTX (radeonsi, navi31, LLVM 20.1....)
vainfo: Supported profile and entrypoints
      VAProfileNone                   : VAEntrypointVideoProc
```
and `mpv` doesn't seem to use vaapi when requested.

For the `drm-renderers` i have no idea how to test them. I'm not even sure if i got it correctly. `amdgpu-experimental` certainly referenced to the amdgpu driver. `msm` should be `freedreno` but i'm not really sure here.

thesamesam reviewed (commented):
Thanks a lot for looking at this!

I'd suggest including the details from the PR description inside the commit message. We want things to be preserved if GH disappears and also to just make `git log` more useful.

Please let @chewi or @tamiko (or anyone else who fancies/knows about this stuff) take a look first before merging. IIRC Chewi has mentioned using this before so picking on him so we have some input here.

mm1ke commented on media-libs/virglrenderer/virglrenderer-9999.ebuild:
```diff
@@ -21,25 +21,31 @@ HOMEPAGE="https://virgil3d.github.io/"
 LICENSE="MIT"
 SLOT="0"
-IUSE="static-libs test"
-# Most of the testsuite cannot run in our sandboxed environment, just don't
```
Sorry, didn't intend to remove them - added them back.

thesamesam commented:
Sorry, I'd forgot about this, my assumption was you'd merge it after the positive comments & no negative ones ;)

mm1ke commented:
> Sorry, I'd forgot about this, my assumption was you'd merge it after the positive comments & no negative ones ;)

heh, not yet. I was hoping to get some feedback from @chewi or @tamiko but actually forgot about this PR too. :)
I've updated the PR and split the commit into two (one for the changes in the live ebuild and one for the updated 1.1.1 ebuild). If everything is fine and i wouldn't get any negative comments i'll push it soon(tm).
